docs/guide/advanced/vue-router.md

```diff
@@ -134,26 +134,32 @@ const Posts = {
 
 The root of the app displays a `<router-link>` leading to `/posts`, where we list the posts.
 
-The real router looks like this:
+The real router looks like this. Notice that we're exporting the routes separately from the route, so that we can instantiate a new router for each individual test later.
 
 ```js
 import { createRouter, createWebHistory } from 'vue-router'
 
+const routes = [
+  {
+    path: '/',
+    component: {
+      template: 'Welcome to the blogging app'
+    }
+  },
+  {
+    path: '/posts',
+    component: Posts
+  }
+];
+
 const router = createRouter({
   history: createWebHistory(),
-  routes: [
-    {
-      path: '/',
-      component: {
-        template: 'Welcome to the blogging app'
-      }
-    },
-    {
-      path: '/posts',
-      component: Posts
-    }
-  ]
+  routes: routes,
 })
+
+export { routes };
```

Comment:
I suppose you exported the routes to reuse them later, but I don't see where.

Reply:
> Nice, thank you for the PR!
>
>In my opinion, we should show more of vue-router-mock, as it simplifies the tests a great deal in real applications. But that's still better, so thanks 👍

Is there a ticket / discussion for this somewhere? I'd be willing to try and create better examples with regards to mocking vue-router, but I'd need some input on what you folks think is good practice. Let me know if I can help!

Reply:
👍

I don't think there is a ticket to track this, but the idea is to use `createRouterMock`.
For example, in a route `/users/:userId` that displays a `User` component:
```ts
const router = createRouterMock();

function userWrapper() {
  injectRouterMock(router);
  // fake router params
  router.setParams({
    userId: 12
  });
  return mount(User);
}
```

and then in the tests, you can easily check when the router has been called to navigate, because `router.push` is spied on:
```ts
expect(router.push).toHaveBeenCalledWith('/');
```

The advantage is that you don't have to export your routes and really trigger a navigation: you just test your component in isolation, set the params you want, and can check on the navigations triggered.

If you want to open another PR, that would be awesome 👍

Reply:
So far I stayed away from vue-router-mock, cause I feel adding more external dependencies on top of other dependencies increases the risk of it not being compatible in case Vue has another major upgrade. In this case we're talking about Vue > Jest > TestUtils and then adding vue-router-mock as another dependency. That's 4 levels deep :)

But I'm willing to at least give other people the option with some better docs, so let me explore that next week :)